test/unit/preprocessing/test_preprocessing.py

```diff
@@ -2,55 +2,45 @@
 #
 # SPDX-License-Identifier: MPL-2.0
 
-import math
 from test.unit.utils.base import BaseTestCase
 
-import numpy as np
 import pandas as pd
 
 from openstef.preprocessing import preprocessing
 
 
 class TestPreprocessing(BaseTestCase):
     def test_replace_repeated_values_with_nan(self):
-        start_nan = 8
-        end_nan = 11
-        data = np.arange(20)
-        df = pd.DataFrame({"Column1": data[:], "Column2": data[:], "Column3": data[:]})
-        # Add 4 sequentially same numbers
-        df["Column2"] = np.where(
-            df["Column2"].between(start_nan, end_nan), 30, df["Column2"]
+        # Arrange
+        data_no_repeated_sequences = [0, 1, 2, 3, 4, 5, 6, 5, 4, 3, 2, 1]
+        data_two_repeated_sequences = [0, 1, 2, 2, 2, 2, 3, 4, 2, 2, 2, 5]
+
+        df = pd.DataFrame(
+            {
+                "Column1": data_no_repeated_sequences,
+                "Column2": data_two_repeated_sequences,
+            }
         )
 
-        # No change, since there are 4 seq. numbers.
-        df_no_repeated = preprocessing.replace_repeated_values_with_nan(
-            df, 5, "Column2"
+        # Act
+        df_no_repeated_sequences = preprocessing.replace_repeated_values_with_nan(
+            df, 2, "Column1"
         )
-        self.assertFalse(df_no_repeated.isna().values.any())
 
-        # No change, since there are 4 seq. numbers.
-        df_no_repeated = preprocessing.replace_repeated_values_with_nan(
+        df_repeated_sequences_below_threshold = (
```

Comment:
Tiny optional improvement would be to split this unit test into two. In principle a single unit test should only test a single test case. Thus the first unit test can test that a sequence with repetitions indeed removes them properly. A second unit test can test that a sequence without repetitions is not changed.